resources/sqlite/src/lib.rs

```diff
@@ -0,0 +1,171 @@
+//!     )
+//!     .execute(&pool)
+//!     .await
+//!     .unwrap();
+//! }
+//! ```
+//!
+//! ## Configuration
+//! The database can be configured using [`ShuttleSqliteConnOpts`] which mirrors sqlx's [`SqliteConnectOptions`](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html) for the
+//! options it exposes.
+//!
+//! Construction of the full connection string is handled internally for security reasons and defaults to creating a
+//! file-based database named `default_db.sqlite` with `create_if_missing == true`. Use the `filename` and/or
+//! `in_memory` methods to configure the type of database created.
+//!
+//! See [`SqliteConnectOptions::new()`](https://docs.rs/sqlx/latest/sqlx/sqlite/struct.SqliteConnectOptions.html#method.new)
+//! for all default settings.
+//!
+//! ```ignore
+//! #[shuttle_runtime::main]
+//! async fn axum(
+//!     #[shuttle_sqlite::ShuttleSqlite(opts = ShuttleSqliteConnOpts::new().filename("custom.sqlite"))] pool: shuttle_sqlite::SqlitePool,
+//! ) -> shuttle_axum::ShuttleAxum { /* ... */ }
+//! ```
+//! Note that Shuttle does currently not support the `collation`, `thread_name`, `log_settings`, `pragma`, `extension`,
+//! `shared_cache` options.
+use async_trait::async_trait;
+use serde::Serialize;
+use shuttle_service::{Factory, ResourceBuilder, Type};
+
+mod conn_opts;
+pub use conn_opts::*;
+
+use sqlx::sqlite::SqliteConnectOptions;
+/// The [`sqlx::SqlitePool`](https://docs.rs/sqlx/latest/sqlx/type.SqlitePool.html) that is being returned to the user.
+///
+pub use sqlx::SqlitePool;
+
+/// Builder struct used to configure the database, e.g. `SQLite(opts = SQLiteConnOpts::new())`.
+#[derive(Serialize)]
+pub struct ShuttleSqlite {
+    opts: ShuttleSqliteConnOpts,
+}
+
+impl ShuttleSqlite {
+    pub fn opts(mut self, opts: ShuttleSqliteConnOpts) -> Self {
+        self.opts = opts;
+        self
+    }
+
+    pub fn in_memory(mut self, on: bool) -> Self {
+        self.opts.in_memory = on;
+        self
+    }
+}
+
+#[async_trait]
+impl ResourceBuilder<sqlx::SqlitePool> for ShuttleSqlite {
+    /// The type of resource this creates
+    const TYPE: Type = Type::EmbeddedDatabase;
+
+    /// The internal config being constructed by this builder. This will be used to find cached [Self::Output].
+    type Config = Self;
+
+    /// The output type used to build this resource later
+    type Output = ShuttleSqliteConnOpts;
+
+    /// Create a new instance of this resource builder
+    fn new() -> Self {
+        Self {
+            opts: ShuttleSqliteConnOpts::default(),
+        }
+    }
+
+    /// Get the internal config state of the builder
+    ///
+    /// If the exact same config was returned by a previous deployement that used this resource, then [Self::output()]
+    /// will not be called to get the builder output again. Rather the output state of the previous deployment
+    /// will be passed to [Self::build()].
```

Comment:
You don't have to repeat comments on a trait method since the comments from the trait will always be used. Ie. these will never be shown to the user and will have to be kept in-sync with the trait comments

Reply:
Didn't know that. I originally just put it there for myself to make sense of the trait.